File: C++ solutions/2521. Distinct Prime Factors of Product of Array.cpp
class Solution {
public:
    set<int>st;
    void primeFact(int n){
        while(n%2==0){
            st.insert(2);
            n/=2;
        }
        int ans=sqrt(n);
        for(int i=3;i<=ans;i+=2){
            while(n%i==0){
                st.insert(i);
                n/=i;
            }
        }
        if(n>2) st.insert(n);
    }
    int distinctPrimeFactors(vector<int>& nums) {
       for(int i=0;i<nums.size();i++){
           primeFact(nums[i]);
       }
       return st.size();
    }
};

File: JavaScript solutions/2521. Distinct Prime Factors of Product of Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var distinctPrimeFactors = function (nums) {
  st = new Set();
  const primeFact = (n) => {
    while (n % 2 == 0) {
      st.add(2);
      n /= 2;
    }
    let ans = Math.sqrt(n);
    for (let i = 3; i <= ans; i += 2) {
      while (n % i == 0) {
        st.add(i);
        n /= i;
      }
    }
    if (n > 2) st.add(n);
  };

  for (let i = 0; i < nums.length; i++) {
    primeFact(nums[i]);
  }
  return st.size;
};
